internal/constraints/operator_num.go

```diff
@@ -0,0 +1,57 @@
+package constraints
+
+import (
+	"github.com/Unleash/unleash-client-go/v3/api"
+	"github.com/Unleash/unleash-client-go/v3/context"
+	"math/big"
+)
+
+func operatorNumEq(ctx *context.Context, constraint api.Constraint) (bool, error) {
+	return operatorNum(ctx, constraint, func(context *big.Float, constraint *big.Float) bool {
+		return context.Cmp(constraint) == 0
+	})
+}
+
+func operatorNumLt(ctx *context.Context, constraint api.Constraint) (bool, error) {
+	return operatorNum(ctx, constraint, func(context *big.Float, constraint *big.Float) bool {
+		return context.Cmp(constraint) < 0
+	})
+}
+
+func operatorNumLte(ctx *context.Context, constraint api.Constraint) (bool, error) {
+	return operatorNum(ctx, constraint, func(context *big.Float, constraint *big.Float) bool {
+		return context.Cmp(constraint) <= 0
+	})
+}
+
+func operatorNumGt(ctx *context.Context, constraint api.Constraint) (bool, error) {
+	return operatorNum(ctx, constraint, func(context *big.Float, constraint *big.Float) bool {
+		return context.Cmp(constraint) > 0
+	})
+}
+
+func operatorNumGte(ctx *context.Context, constraint api.Constraint) (bool, error) {
+	return operatorNum(ctx, constraint, func(context *big.Float, constraint *big.Float) bool {
+		return context.Cmp(constraint) >= 0
+	})
+}
+
+func operatorNum(
+	ctx *context.Context,
+	constraint api.Constraint,
+	check func(context *big.Float, constraint *big.Float) bool,
+) (bool, error) {
+	contextValue := ctx.Field(constraint.ContextName)
+	contextParsed, _, contextErr := big.ParseFloat(contextValue, 10, 0, big.ToNearestEven)
+	constraintParsed, _, constraintErr := big.ParseFloat(constraint.Value, 10, 0, big.ToNearestEven)
```

Comment:
This one is interesting, I know we in rust did equal as absolute difference between a and b less than :epsilon: which for rust's f64 is `2.2204460492503131*10^-16` or `2^-52` - as in, rust does not do rounding. Neither does javascript, which treats it as a Number which also does equivalence as `absdiff < Number.EPSILON`.

Reply:
For current client specifications which only tests that the sdk can handle floats, not how they handle difference this is ok.

Reply:
Yeah, wasn't sure about this one. This seemed like a sensible, built-in way to compare floats, but would be nice with a common approach.